src/components/TubeMapContainer.js

```diff
@@ -141,36 +141,52 @@ class TubeMapContainer extends Component {
         const error = "Fetching remote data returned error";
         throw new Error(error);
       } else {
-        const nodes = tubeMap.vgExtractNodes(json.graph);
-        const tracks = tubeMap.vgExtractTracks(json.graph);
-        // Call vgExtractReads on each individual read and store in readsArr
-        let readsArr = [];
-        for (const gam of Object.values(json.gam)) {
-          // Include readsArr length to prevent duplicate ids
-          readsArr.push(tubeMap.vgExtractReads(nodes, tracks, gam, readsArr.length));
-        }
-        
         // go through viewTarget and create array of read file track numbers
-        let sourceTrackIDs = [];
+        let readTrackIDs = [];
+        // And the graph track number if any
+        let graphTrackID = null;
+        // And the haplotype track number if any
+        let haplotypeTrackID = null;
         for (let i = 0; i < this.props.viewTarget.tracks.length; i++) {
           const track = this.props.viewTarget.tracks[i];
-          //add track index to array if the track contains a gam file
           for (const file of track.files) {
             if (file.type === "read") {
-              sourceTrackIDs.push(i);
+              //add track index to array if the track contains a gam file
+              readTrackIDs.push(i);
+              break;
+            }
+            if (file.type === "graph") {
+              // Or note if it is a graph (one allowed)
+              graphTrackID = i;
+              break;
+            }
+            if (file.type === "haplotype") {
+              // Or a collection of haplotypes (one allowed)
+              haplotypeTrackID = i;
               break;
             }
           }
         }
 
-        console.log(sourceTrackIDs);
-        // Go through every read and assign it a source file number
-        for (let i = 0; i < readsArr.length; i++) {
-          for (let j = 0; j < readsArr[i].length; j++) {
-            readsArr[i][j].sourceTrackID = sourceTrackIDs[i];
-          }
-        }
+        console.log("Graph track: " + graphTrackID + " Haplotype track: " + haplotypeTrackID);
+
+        const nodes = tubeMap.vgExtractNodes(json.graph);
+        // TODO: For now claim haplotypes came from the graph
```

Comment:
I'm not sure if I understand what needs to be done here. Do haplotypes need to be extracted separately?